sounds/views.py

```diff
@@ -318,22 +318,13 @@ def sound_download(request, username, sound_id):
     if sound.user.username.lower() != username.lower():
         raise Http404
 
-    if settings.LOG_DOWNLOADS:
-        range_header = request.META.get('HTTP_RANGE', '').replace('bytes=', '').split('-')
-        if 'HTTP_RANGE' not in request.META or range_header[0] == "0":
-            downloads_logger.info('Download sound', extra={
-                'user_id': request.user.id,
-                'user_ip': request.META.get('HTTP_X_FORWARDED_FOR'),
-                'protocol': request.META.get('HTTP_X_FORWARDED_PROTOCOL'),
-                'session_id': request.session.session_key,
-                'user_agent': request.META.get('HTTP_USER_AGENT'),
-                'method': request.method,
-                'sound_id': sound_id,
-                'range': request.META.get('HTTP_RANGE', None),
-            })
-
-    if not Download.objects.filter(user=request.user, sound=sound).exists():
-        Download.objects.create(user=request.user, sound=sound, license=sound.license)
+    if 'HTTP_RANGE' not in request.META:
```

Comment:
Can we have an explanation here:

> Download managers and some browsers use the `range` header to download files in multiple parts. We have observed that all clients first make a GET with no `range` header (to get the file length) and then make multiple other requests. We ignore all requests that have `range` header because we assume that a first query has already been made.
We additionally guard against users clicking on download multiple times by storing a sentinel in the cache for 5 minutes.